### PR TITLE
[netcore] Ongoing work.

### PR DESCRIPTION
**Pull request by vargaz**
* Implement all CustomAttributeExtensions methods.
* Add stubs for __ComObject which is needed by the runtime.

### CODE REVIEW

File: mcs/class/System.Private.CoreLib/System/__ComObject.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace System
+{
+	internal class __ComObject
```
Comment (marek-safar) [inline mcs/class/System.Private.CoreLib/System/__ComObject.cs]: Is it really needed?

Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System/__ComObject.cs]: The runtime asserts on it.

Comment (marek-safar) [reply mcs/class/System.Private.CoreLib/System/__ComObject.cs]: Can we remove it? Otherwise, we'll have to teach linker etc which might be even more work

Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System/__ComObject.cs]: Its needed for COM support, we can get rid of it, but its more work than adding this one class.

**Pull request merged**